util/patch/patch.go

```diff
@@ -238,9 +322,9 @@ func (h *Helper) patchStatusConditions(ctx context.Context, obj client.Object, f
 
 	// Start the backoff loop and return errors if any.
 	return wait.ExponentialBackoff(backoff, func() (bool, error) {
-		latest, ok := before.DeepCopyObject().(conditions.Setter)
+		latest, ok := h.beforeObject.DeepCopyObject().(client.Object)
 		if !ok {
-			return false, errors.Errorf("object %s doesn't satisfy conditions.Setter, cannot patch", latest.GetObjectKind())
+			return false, errors.Errorf("%s %s doesn't satisfy conditions.Setter, cannot patch", h.gvk.Kind, latest.GetObjectKind())
```

Comment:
```suggestion
			return false, errors.Errorf("%s %s doesn't satisfy client.Object, cannot patch", h.gvk.Kind, klog.KObj(h.beforeObject))
```

This one was missed (deleted my previous comment to reduce noise here)

Reply:
fixed